File: app/src/main/java/com/galib/natorepbs2/ui/ComplainCentreActivity.java
package com.galib.natorepbs2.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.galib.natorepbs2.viewmodel.ComplainCentreViewModel;
import com.galib.natorepbs2.R;

public class ComplainCentreActivity extends AppCompatActivity {
    ComplainCentreViewModel complainCentreViewModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_complain_centre);
        complainCentreViewModel = new ViewModelProvider(this).get(ComplainCentreViewModel.class);
        RecyclerView recyclerView = findViewById(R.id.recyclerview);
        final ComplainCentreAdapter adapter = new ComplainCentreAdapter(this, new ComplainCentreAdapter.WordDiff());
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        complainCentreViewModel.getAllComplainCentre().observe(this, adapter::submitList);
    }
}

File: app/src/main/java/com/galib/natorepbs2/ui/ComplainCentreActivity.kt
package com.galib.natorepbs2.ui

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.galib.natorepbs2.NPBS2Application
import com.galib.natorepbs2.R
import com.galib.natorepbs2.viewmodel.ComplainCentreViewModel
import com.galib.natorepbs2.viewmodel.ComplainCentreViewModelFactory

class ComplainCentreActivity : AppCompatActivity() {
    private val complainCentreViewModel: ComplainCentreViewModel by viewModels {
        ComplainCentreViewModelFactory((application as NPBS2Application).repository)
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_complain_centre)
        val recyclerView = findViewById<RecyclerView>(R.id.recyclerview)
        val adapter = ComplainCentreAdapter(this, ComplainCentreAdapter.WordDiff())
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)
        complainCentreViewModel.allComplainCentre.observe(this) { list ->
            adapter.submitList(
                list
            )
        }
    }
}